src/commons/api/ca.rs

```diff
@@ -1574,6 +1571,15 @@ pub enum ExchangeResult {
     Failure(ErrorResponse),
 }
 
+impl ExchangeResult {
+    pub fn was_success(&self) -> bool {
+        match self {
```

Comment:
I'm not sure it's better but I think you could shrink this to something like `matches!(self, ExchangeResult::Success)`.